packages/apollo-gateway/src/index.ts

```diff
@@ -84,6 +104,15 @@ export class ApolloGateway implements GraphQLService {
       this.createSchema(config.localServiceList);
     }
 
+    if (isHostedConfig(config)) {
+      const apiKey = config.apiKey || process.env['ENGINE_API_KEY'];
```

Comment:
Not asking you to fix it unless you feel so compelled, but just pointing out that I think we have this logic duplicated in (too?) many places now!

Reply:
Went ahead and did a `config.apiKey = config.apiKey || process.env['ENGINE_API_KEY']` in the constructor.